services/core/java/com/android/server/clipboard/ClipboardService.java

```diff
@@ -429,9 +430,7 @@ private void scheduleAutoClear(@UserIdInt int userId, int intendingUid) {
         }
 
         private long getTimeoutForAutoClear() {
```

Comment:
Seems like this isn't a per-user Context (and mContext seems invalid here, build error) which will cause the settings in the Owner profile to apply across all profiles irrespective of the current profile's setting, so you'll need to specify the uid this is for.

```suggestion
        private long getTimeoutForAutoClear(int userId) {
            return ExtSettings.CLIPBOARD_AUTO_CLEAR_TIMEOUT.get(getContext(), userId);
```

And where `getTimeoutForAutoClear` method is called, pass the `userId` argument:

`getTimeoutForAutoClear(userId)`

Reply:
Didn't catch that build error as I'm currently using `m Settings`. Waiting for a new SSD to arrive (don't have enough space on device).